Makefile

```diff
@@ -81,12 +98,32 @@ distclean:
 	-rm -rf deps/icu
 	-rm -rf deps/icu4c*.tgz deps/icu4c*.zip deps/icu-tmp
 	-rm -f $(BINARYTAR).* $(TARBALL).*
+	-rm -rf deps/v8/testing/gmock
+	-rm -rf deps/v8/testing/gtest
 
 check: test
 
 cctest: all
 	@out/$(BUILDTYPE)/$@
 
+v8:
+	tools/make-v8.sh v8
+ifneq (,$(filter $(DESTCPU),x86))
+	+$(MAKE) -C deps/v8 $(V8_BUILD_OPTIONS)
+else
+ifneq (,$(filter $(ARCH),x86))
+	+$(MAKE) -C deps/v8 $(V8_BUILD_OPTIONS)
+else
+ifeq ($(ARCH)$(DESTCPU),)
+	+$(MAKE) -C deps/v8 $(V8_BUILD_OPTIONS)
+else
+	+$(MAKE) -C deps/v8 $(ARCH) $(V8_BUILD_OPTIONS)
+endif
+	+$(MAKE) -C deps/v8 $(ARCH) $(V8_BUILD_OPTIONS)
+endif
+	+$(MAKE) -C deps/v8 $(ARCH) $(V8_BUILD_OPTIONS)
```

Comment:
When I'm performing tests, I personally prefer to pass $DESTCPU to the makefile, but the makefile uses $DESTCPU to determine $ARCH. The reason both arch and destcpu are there is because I'm a bit paranoid that sometimes one or the other won't be defined. In the case that both of these vars are defined, ARCH is preferred.

I don't remember why I put the plus signs there, but IIRC it may be related to the fact that there's a recursive make call. I'm not sure though.